src/ejabberd_auth.erl

```diff
@@ -524,61 +524,76 @@ db_set_password(User, Server, Password, Mod) ->
 
 db_get_password(User, Server, Mod) ->
     UseCache = use_cache(Mod, Server),
-    case erlang:function_exported(Mod, get_password, 2) of
-	false when UseCache ->
-	    ets_cache:lookup(?AUTH_CACHE, {User, Server});
-	false ->
-	    error;
-	true when UseCache ->
-	    ets_cache:lookup(
-	      ?AUTH_CACHE, {User, Server},
-	      fun() -> Mod:get_password(User, Server) end);
-	true ->
-	    Mod:get_password(User, Server)
+    case Mod == ejabberd_auth_http of
```

Comment:
Yep. There was no other good way to fork the flow. Sometimes the check done in the parent case would explode if evaluated, so I needed to pre-empt that.